src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -248,6 +248,18 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None):
                 data['medflux'].attrs['flux_units'] = \
                     data.flux.attrs['flux_units']
 
+                # correct G395H curvature
+                if meta.inst == 'nirspec' and data.mhdr['GRATING'] == 'G395H':
+                    if meta.curvature == 'correct':
+                        log.writelog('  In NIRSpec G395H setting:', mute=(not meta.verbose))
+                        log.writelog('  Correcting curvature and bringing trace in the center '
+                                     'of the detector', mute=(not meta.verbose))
+                        log.writelog('  !!! Ensure that you are using meddata for the optimal '
+                                     'extraction profile !!!', mute=(not meta.verbose))
+                        log.writelog('  Updating src_ypos and median frame '
+                                     'accordingly.', mute=(not meta.verbose))
+                        data, meta = inst.straighten_trace(data, meta)
```

Comment:
Can you pass in the log and replace the print statements in `straighten_trace` with logs?

Reply:
Yes! I updated this, and at the same time I also moved most of the logs to the straighten_trace() function. It makes s3_reduce.py look nicer.